test/e2e/tests/single_zone_e2e_test.go

```diff
@@ -294,6 +302,163 @@ var _ = Describe("GCE PD CSI Driver", func() {
 		}()
 	})
 
+	It("Should create CMEK key, go through volume lifecycle, validate behavior on key revoke and restore", func() {
+		ctx := context.Background()
+		Expect(testContexts).ToNot(BeEmpty())
+		testContext := getRandomTestContext()
+
+		controllerInstance := testContext.Instance
+		controllerClient := testContext.Client
+
+		p, z, _ := controllerInstance.GetIdentity()
+		locationID := "global"
+
+		// The resource name of the key rings.
+		parentName := fmt.Sprintf("projects/%s/locations/%s", p, locationID)
+		keyRingId := "gce-pd-csi-test-ring"
+
+		// Create KeyRing
+		ringReq := &kmspb.CreateKeyRingRequest{
+			Parent:    parentName,
+			KeyRingId: keyRingId,
+		}
+		keyRing, err := kmsClient.CreateKeyRing(ctx, ringReq)
+		if !gce.IsGCEError(err, "alreadyExists") {
+			getKeyRingReq := &kmspb.GetKeyRingRequest{
+				Name: fmt.Sprintf("%s/keyRings/%s", parentName, keyRingId),
+			}
+			keyRing, err = kmsClient.GetKeyRing(ctx, getKeyRingReq)
+
+		}
+		Expect(err).To(BeNil(), "Failed to create or get key ring %v", keyRingId)
+
+		// Create CryptoKey in KeyRing
+		keyId := "test-key-" + string(uuid.NewUUID())
+		keyReq := &kmspb.CreateCryptoKeyRequest{
+			Parent:      keyRing.Name,
+			CryptoKeyId: keyId,
+			CryptoKey: &kmspb.CryptoKey{
+				Purpose: kmspb.CryptoKey_ENCRYPT_DECRYPT,
+				VersionTemplate: &kmspb.CryptoKeyVersionTemplate{
+					Algorithm: kmspb.CryptoKeyVersion_GOOGLE_SYMMETRIC_ENCRYPTION,
+				},
+			},
+		}
+		key, err := kmsClient.CreateCryptoKey(ctx, keyReq)
+		Expect(err).To(BeNil(), "Failed to create crypto key %v in key ring %v", keyId, keyRing.Name)
+
+		keyVersions := []string{}
+		keyVersionReq := &kmspb.ListCryptoKeyVersionsRequest{
+			Parent: key.Name,
+		}
+
+		it := kmsClient.ListCryptoKeyVersions(ctx, keyVersionReq)
+
+		for {
+			keyVersion, err := it.Next()
+			if err == iterator.Done {
+				break
+			}
+			Expect(err).To(BeNil(), "Failed to list crypto key versions")
+
+			keyVersions = append(keyVersions, keyVersion.Name)
+		}
+
+		// Defer deletion of all key versions
+		// https://cloud.google.com/kms/docs/destroy-restore
+		// Temporarily disable revokation of CMEK key because of test infra CI permissions
```

Comment:
this is still not resolved? will we be leaking resources?

Reply:
I'll add the permission for this. we are already leaking keyrings by design anyway (keyrings don't support deletion).